File: LC 1723.cpp
class Solution {
public:
    int minimumTimeRequired(vector<int>& jobs, int k) {
        int n = jobs.size();
        vector<int> sum(1 << n);
        for (int i = 0; i < n; ++i) {
            for (int j = 0, bit = (1 << i); j < bit; ++j) {
                sum[bit|j] = sum[j] + jobs[i];
            }
        }

        vector<int> dp(sum);
        for (int i = 1; i < k; ++i) {
            for (int j = (1 << n) - 1; j > 0; --j) {
                for (int s = j; s > 0; s = (s - 1) & j) {
                    dp[j] = min(dp[j], max(sum[s], dp[j ^ s]));
                }
            }
        }
        return dp.back();
    }
};

File: 2000/LC 1723.go
func minimumTimeRequired(jobs []int, k int) int {
    n := len(jobs)
    sort.Ints(jobs)
    sum := make([]int, 1 << n)
    for i := range jobs {
        for j, s := 0, 1 << i; j < s; j++ {
            sum[s|j] = sum[j] + jobs[i]
        }
    }

    f := make([]int, 1 << n)
    copy(f, sum)
    for i := 1; i < k; i++ {
        for j := (1 << n) - 1; j > 0; j-- {
            for s := j; s > 0; s = (s - 1) & j {
                f[j] = min(f[j], max(sum[s], f[s ^ j]))
            }
        }
    }

    return f[len(f)-1]
}